pkg/services/datasources/service/datasource.go

```diff
@@ -321,12 +321,8 @@ func (s *Service) prepareInstanceSettings(ctx context.Context, settings *backend
 		return settings, nil
 	}
 
-	// When the APIVersion is set, the client must also implement AdmissionHandler
-	if p.APIVersion == "" {
-		if settings.APIVersion != "" {
-			return nil, fmt.Errorf("invalid request apiVersion (datasource does not have one configured)")
-		}
-		return settings, nil // NOOP
+	if settings.APIVersion == "" {
```

Comment:
Note that we cannot merge this as it is because Prometheus and Graphite exposes API Servers but don't define a `Validation` or `Mutation` hooks, which are required below. If we want to go this route, we need to define them.